### PR TITLE
fix(node-resolve): allow deduplicating custom module dirs

### PR DESCRIPTION
**Pull request by LarsDenBakker**
## Rollup Plugin Name: `node-resolve`

This PR contains:

- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

List any relevant issue numbers: fixes https://github.com/rollup/plugins/issues/100

### Description

There is no need to specify the module directory when deduplicating, by changing the base dir you can have the same effect while allowing the user to customize the module directory.

### CODE REVIEW

File: packages/node-resolve/src/index.js [hunk 1]
```diff
@@ -295,11 +295,8 @@ export default function nodeResolve(options = {}) {
       // ignore IDs with null character, these belong to other plugins
       if (/\0/.test(importee)) return null;
 
-      const basedir = importer ? dirname(importer) : process.cwd();
-
-      if (shouldDedupe(importee)) {
-        importee = join(process.cwd(), 'node_modules', importee);
-      }
+      const rootDir = process.cwd();
+      const basedir = importer && !shouldDedupe(importee) ? dirname(importer) : rootDir;
```
Comment (NotWoods) [inline packages/node-resolve/src/index.js]: I think `!shouldDedupe` is a little hard to read, and `importer != null` is a bit easier to grok. Potentially rewrite the condition as:
```suggestion
      const basedir = !importer || shouldDedupe(importee) ? dirname(importer) : rootDir;
```

Comment (LarsDenBakker) [reply packages/node-resolve/src/index.js]: Makes sense!

**Comment by shellscape** [comment]
@LarsDenBakker I noticed two changes to `auto-install` files. Can you comment on why those were changed?

**Comment by LarsDenBakker** [comment]
@shellscape I'm not sure how that happened, when I run `pnpm run lint` from the root of the project some more files get modified as well. I've reset and undid those changes.

**Comment by shellscape** [comment]
that's interesting. I'll have a look from my end to resolve that.

**Comment by LarsDenBakker** [comment]
Looks like you already fixed those on master today, I didn't sync that yet.

**Comment by shellscape** [comment]
No worries at all. Big repo, this is bound to happen now and then :beer:

**Comment by shellscape** [comment]
@LarsDenBakker looks like your other PR created a conflict with this one. Please have a look.

**Comment by LarsDenBakker** [comment]
I've resolved the conflicts

**Comment by shellscape** [comment]
@rollup/plugin-contributors @rollup/plugin-maintainers please give this one a look

**Comment by shellscape** [comment]
I'm going to keep this open for another day or so, so that Github can use it as an example to fix a bug with PR approvals.

**Pull request merged**